### PR TITLE
Refactored summary cli methods

### PR DESCRIPTION
**Pull request by mortonjt**
This changes the CLI so that visualizations are dumped within the regression method.

Ultimately, this removes the need to create intermediate q2 types and files.

### CODE REVIEW

File: gneiss/plot/_regression_plot.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+                     header=True, index=True)
+    pvalues = model.pvalues
+    pvalues.to_csv(os.path.join(output_dir, 'pvalues.csv'),
+                   header=True, index=True)
+    balances = model.balances
+    balances.to_csv(os.path.join(output_dir, 'balances.csv'),
+                    header=True, index=True)
+
+
+def _deposit_results_html(index_f):
```
Comment (qiyunzhu) [inline gneiss/plot/_regression_plot.py:239]: This function doesn't have much docstring. What is the datatype of `index_f`? Is it a file handle? Why the `write` function has to have two nesting pairs of parentheses? I guess that this function can me much simpler.

File: gneiss/plot/_regression_plot.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    p3 = _projected_residuals(model, plot_width=w, plot_height=h)
+    hm_p = _heatmap_summary(model.pvalues, model.coefficients())
+
+    # combine the cross validation, explained sum of squares tree and
+    # residual plots into a single plot
+    p = row(column(p2, p3), p1)
+    p = column(hm_p, p)
+    index_fp = os.path.join(output_dir, 'index.html')
+    with open(index_fp, 'w') as index_f:
+        index_f.write('<html><body>\n')
```
Comment (qiyunzhu) [inline gneiss/plot/_regression_plot.py:313]: So the HTML page does not have `<head>`. Although not required, it is favorable to put things like `<title>` and external resources in it.

File: gneiss/plot/_regression_plot.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+        index_f.write(smry.as_html())
+        _deposit_results_html(index_f)
+
+        plot_html = file_html(p, CDN, 'Diagnostics')
+        index_f.write(plot_html)
+        index_f.write('</body></html>\n')
+
+
+# LME summary
+def lme_summary(output_dir: str, model: LMEModel, tree: TreeNode) -> None:
```
Comment (qiyunzhu) [inline gneiss/plot/_regression_plot.py:324]: I didn't know this way of defining a function :)

File: gneiss/plot/tests/test_regression_plot.py [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+        self.tree = TreeNode.read(['(c, (b,a)Y2)Y1;'])
+        self.metadata = pd.DataFrame({
+            'lame': [1, 1, 1, 1, 1],
+            'real': [1, 2, 3, 4, 5]
+        }, index=['s1', 's2', 's3', 's4', 's5'])
+
+        np.random.seed(0)
+        n = 15
+        a = np.array([1, 4.2, 5.3, -2.2, 8])
+        x1 = np.linspace(.01, 0.1, n)
```
Comment (qiyunzhu) [inline gneiss/plot/tests/test_regression_plot.py:47]: Is there a preference of using `.01` or `0.01`?

**Review by qiyunzhu — COMMENTED** [review]
Very good job! Looks to me that many lines were moved from `_plot.py` to `_regression_plot.py`. So it is more of a fine adjustment instead of new functionality, right?

File: gneiss/plot/tests/test_regression_plot.py (hunk 4 above)
Comment (mortonjt) [reply gneiss/plot/tests/test_regression_plot.py:47]: Not really.

**Comment by mortonjt** [comment]
Comments addressed!  Thanks @qiyunzhu for looking over this!  And yes, I'd love to hear about how to refactor the html results table.

File: gneiss/plot/_regression_plot.py (hunk 2 above)
Comment (mortonjt) [reply gneiss/plot/_regression_plot.py:313]: 👍  let's sit down and figure out how to make this better.

File: gneiss/plot/_regression_plot.py (hunk 1 above)
Comment (mortonjt) [reply gneiss/plot/_regression_plot.py:239]: Yes, its a filehandle.  The nested parentheses is for concatenating strings. I updated the docstring. Let me know what you think!

**Comment by qiyunzhu** [comment]
+1. You are welcome! Let's chat next week on the HTML thing. @mortonjt

**Pull request merged**